x/ecocredit/basket/date_criteria.go

```diff
@@ -17,3 +19,34 @@ func (d *DateCriteria) ToApi() *api.DateCriteria {
 	}
 	return nil
 }
+
+// Validate validates DateCriteria, ensuring only one field is set,
+// minimum start date is after 1900-01-01, start date window is at
+// least one day, and years in the past is at least one year
+func (d *DateCriteria) Validate() error {
```

Comment:
Moved from `msg_create.go` to `date_criteria.go` without changes to the underlying logic.